src/chainstate/stacks/transaction.rs

```diff
@@ -248,6 +250,7 @@ impl TransactionPayload {
     }
 
     pub fn new_smart_contract(name: &str, contract: &str) -> Option<TransactionPayload> {
+        increment_stx_smart_contracts(); //promserver
```

Comment:
This method I don't think is used anywhere besides the test code and client-side tooling.  Are you trying to count the number of smart contracts?  If so, then what you should probably do is look at the event receipts emitted in `process_ready_blocks()` in the chains coordinator in order to see which smart contracts get instantiated (and just count those events).

Reply:
Moved to `src/chainstate/coordinator/mod.rs`